ssl/ssl_rsa.c

```diff
@@ -1009,8 +1009,9 @@ int SSL_CTX_use_serverinfo_file(SSL_CTX *ctx, const char *file)
         extension = NULL;
     }
 
-    ret = SSL_CTX_use_serverinfo_ex(ctx, version, serverinfo,
-                                    serverinfo_length);
+    if (version != 0)
```

Comment:
Have you read the `for` loop?  Essentially, there should be no way `version == 0` would reach this point.